libraries/AP_Scripting/applets/arming-checks.lua

```diff
@@ -0,0 +1,440 @@
+REFRESH_RATE = 1000
+INITIAL_DELAY = 20000   -- wait 20 seconds for the AP to settle down before starting to show messages
+
+ALTITUDE_LEGAL_MAX = 120 -- max legal altitude is 120m by default, change if different limits apply
+
+MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7, NONE=-1}
+
+local PARAM_TABLE_KEY = 122
+local PARAM_TABLE_PREFIX = "ZAR_"
+
+local param_idx = 1
+
+local validate      -- forward definition of the validate() function
+
+-- create parameter table for ZAR_ parameters that will be created below
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 25), 'could not add param table')
+
+local arm_auth_id = arming:get_aux_auth_id()
+if arm_auth_id == nil then
+    gcs:send_text(MAV_SEVERITY.ERROR, string.format("%s %s Failed", SCRIPT_NAME, SCRIPT_VERSION) )
+    gcs:send_text(MAV_SEVERITY.ERROR, string.format("%s: get_aux_auth_id failed", SCRIPT_NAME_SHORT))
+    return -- don't execute the script if we can't get an auth_id
+end
+
+----  CLASS: Arming_Check  ----
+local Arming_Check = {}
+Arming_Check.__index = Arming_Check
+
+setmetatable(Arming_Check, {
+    __call = function(cls, func, param_name, text, severity, passed, changed) -- constructor
+        local self      = setmetatable({}, cls)
+        self.func       = func             -- func() is called to validate arming
+        self.param_name = param_name       -- the name of a ZAR_ parameter overriding the severity
+        self.text       = text             -- the message on failure
+        self.severity   = severity         -- is failure a warning or hard stop?
+        self.passed     = passed or nil
+        self.changed    = changed or true
+        return self
+    end
+})
+
+function Arming_Check:state()
+    local passed_new = self.func()
+
+    if self.passed == passed_new then
+        self.changed = false
+    else
+        self.passed = passed_new
+        self.changed = true
+    end
+    return passed_new
+end
+
+FENCE_TYPE = Parameter("FENCE_TYPE")
+FENCE_TOTAL = Parameter("FENCE_TOTAL")
+FENCE_ENABLE = Parameter("FENCE_ENABLE")
+FENCE_AUTOENABLE = Parameter("FENCE_AUTOENABLE")
+FOLL_ENABLE = Parameter("FOLL_ENABLE")
+FOLL_SYSID = Parameter("FOLL_SYSID")
+if FWVersion:type() == 2 then -- Copter specific paramters
+    RTL_ALT = Parameter("RTL_ALT")
+elseif FWVersion:type() == 3 then -- Plane specific Parameters
+    RTL_ALTITUDE = Parameter("RTL_ALTITUDE")
+    RTL_CLIMB_MIN = Parameter("RTL_CLIMB_MIN")
+    Q_ENABLE = Parameter("Q_ENABLE")
+    AIRSPEED_STALL = Parameter("AIRSPEED_STALL")
+    AIRSPEED_MIN = Parameter("AIRSPEED_MIN")
+    AIRSPEED_CRUISE = Parameter("AIRSPEED_CRUISE")
+    AIRSPEED_MAX = Parameter("AIRSPEED_MAX")
+    SCALING_SPEED = Parameter("SCALING_SPEED")
+end
+
+-- this is the pre 4.7 name for this parameter
+local function try_sysid_thismav()
+    MAV_SYSID = Parameter("SYSID_THISMAV")
+end
+if not pcall(try_sysid_thismav) then
+    MAV_SYSID = Parameter("MAV_SYSID")
+end
```

Comment:
If you init without a name you don't get the error if the param is not found, so you don't need the pcall. 
```
MAV_SYSID = Parameter()
if not MAV_SYSID:init("MAV_SYSID") then
    assert(MAV_SYSID:init("SYSID_THISMAV"), "SYSID not found")
end
```

Result is the same, just this might be a little neater.

Reply:
done - thanks, I didn't know that.